### PR TITLE
[MDB IGNORE] [IDB IGNORE] Condensing clothing state modifiers into a decl system.

### PR DESCRIPTION
**Pull request by MistakeNot4892**
## Description of changes
- All the various clothing states (open, buttons, sleeves, tucked in, hood, etc) have been unified in a decl system.
- Accessory hiding vars have been rolled into a check against the state decls.
- Clothing icon validation has been expanded to check for onmob states based on clothing state modifiers.

Depends on #3805 #3806 

## Why and what will this PR improve
Consistent clothing behavior, shorter typetree, more flexible clothing code.

## TODO
- [X] Coarse implementation/conversion.
- [x] Get suits and accessories passing CI.
- [x] Work out why verbs aren't being removed on accessory removal.
- [x] Verify icon updating and clothes icon updating is working on toggle.
- [x] See what happens to unit test time if we test states for all clothing.
- [x] Get all the other clothing passing CI.

## Authorship
Myself.

## Changelog
Nothing player-facing hopefully.

### CODE REVIEW

**Comment by MistakeNot4892** [comment]
Somehow testing every single clothing subtype for icon states didn't appreciably impact testing time. Perplexing.

File: mods/species/bayliens/skrell/_skrell.dm [hunk 1]
```diff
@@ -1,5 +1,4 @@
 #define SPECIES_SKRELL "Skrell"
-#define BODYTYPE_SKRELL "skrellian body"
```
Comment (comma) [inline mods/species/bayliens/skrell/_skrell.dm:2]: Why remove this?

Comment (MistakeNot4892) [reply mods/species/bayliens/skrell/_skrell.dm:2]: It was not used correctly. Didn't do anything, states were all broken as a result.

Comment (comma) [reply mods/species/bayliens/skrell/_skrell.dm:2]: But you're using the same value, just plain string now instead of define, what's different?

Comment (MistakeNot4892) [reply mods/species/bayliens/skrell/_skrell.dm:2]: It's not a `BODYTYPE_` value, using that would be confusing (which I guess is why all the icon states used it) - I can make it a new define if you want but it's only used in like four places so I'd probably just swap the uses to `species.name == SPECIES_SKRELL` or something.

Comment (MistakeNot4892) [reply mods/species/bayliens/skrell/_skrell.dm:2]: Went ahead and did that.

**Review by out-of-phaze — APPROVED** [review]

**Pull request merged**